tests/irdl/test_operation_definition.py

```diff
@@ -168,89 +167,6 @@ def test_attr_verify():
         op.verify()
 
 
-# TODO: remove this test once the Annotated API is deprecated
-@irdl_op_definition
-class ConstraintVarOp(IRDLOperation):
-    name = "test.constraint_var_op"
-
-    T = Annotated[IntegerType | IndexType, ConstraintVar("T")]
-
-    operand = operand_def(T)
-    result = result_def(T)
-    attribute = attr_def(T)
```

Comment:
I would propose keeping the test around while suppressing the deprecation warnings until we delete the mechanism, because things should work even if they're deprecated.

```py
    with pytest.deprecated_call():
```